bertopic/representation/_llamacpp.py

```diff
@@ -150,33 +173,31 @@ def extract_topics(
             self.prompts_.append(prompt)
 
             # Extract result from generator and use that as label
-            topic_description = self.model(prompt, **self.pipeline_kwargs)["choices"]
-            topic_description = [(description["text"].replace(prompt, ""), 1) for description in topic_description]
-
-            if len(topic_description) < 10:
-                topic_description += [("", 0) for _ in range(10 - len(topic_description))]
-
-            updated_topics[topic] = topic_description
+            # topic_description = self.model(prompt, **self.pipeline_kwargs)["choices"]
+            topic_description = self.model.create_chat_completion(
+                  messages=[
+                      {"role": "system", "content": self.system_prompt},
+                      {
+                          "role": "user",
+                          "content": prompt
+                      }
+                  ], ** self.pipeline_kwargs
+            )
+            label = topic_description["choices"][0]["message"]["content"].strip().replace("Topic name: ", "")
+            updated_topics[topic] = [(label, 1)] + [("", 0) for _ in range(9)]
 
         return updated_topics
 
     def _create_prompt(self, docs, topic, topics):
         keywords = ", ".join(list(zip(*topics[topic]))[0])
 
-        # Use the default prompt and replace keywords
-        if self.prompt == DEFAULT_PROMPT:
-            prompt = self.prompt.replace("[KEYWORDS]", keywords)
-
-        # Use a prompt that leverages either keywords or documents in
-        # a custom location
-        else:
-            prompt = self.prompt
-            if "[KEYWORDS]" in prompt:
-                prompt = prompt.replace("[KEYWORDS]", keywords)
-            if "[DOCUMENTS]" in prompt:
-                to_replace = ""
-                for doc in docs:
-                    to_replace += f"- {doc}\n"
-                prompt = prompt.replace("[DOCUMENTS]", to_replace)
+        prompt = self.prompt
+        if "[KEYWORDS]" in prompt:
+            prompt = prompt.replace("[KEYWORDS]", keywords)
+        if "[DOCUMENTS]" in prompt:
+            to_replace = ""
+            for doc in docs:
+                to_replace += f"- {doc}\n"
+            prompt = prompt.replace("[DOCUMENTS]", to_replace)
```

Comment:
If we update this piece of code here, shouldn't we also do this for the other models, like `_cohere.py`?

Reply:
I think it is a bug in the old implementation which does not replace `[DOCUMENTS]` for `DEFAULT_PROMPT`. I have replaced `_create_prompt` in `_llamacpp.py` as the one in `_cohere.py`